manifests/provider/ad.pp

```diff
@@ -139,6 +139,8 @@
   Boolean                                                    $ldap_use_tokengroups                     = true,
   Optional[String[1]]                                        $ldap_group_objectsid                     = undef,
   Optional[String[1]]                                        $ldap_user_objectsid                      = undef,
+  Optional[String[1]]                                        $ldap_user_extra_attrs                    = 'altSecurityIdentities',
+  Optional[String[1]]                                        $ldap_user_ssh_public_key                 = 'altSecurityIdentities',
```

Comment:
I'm OK adding these in as optional parameters, but Optional parameters should always be set to `undef` otherwise, they're not optional.

Reply:
Ok, great! Since this should be the default value, should we just set them as String type instead of Optional[String]?

Reply:
@CodePhase not everyone configures pubkey ssh in AD, so I'd leave them as Optional[String] and document the common values in an example.